subworkflows/nf-scil/rbx/main.nf

```diff
@@ -0,0 +1,98 @@
+import java.util.zip.*
+import org.apache.commons.io.FileUtils
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.zip.ZipFile
+
+include { REGISTRATION_ANTS                 } from '../../../modules/nf-scil/registration/ants/main'
+include { BUNDLE_RECOGNIZE                  } from '../../../modules/nf-scil/bundle/recognize/main'
+
+def fetch_rbx_atlas(atlasUrl, configUrl, dest) {
+
+    def atlas = new File("$dest/atlas.zip").withOutputStream { out ->
+        new URL(atlasUrl).withInputStream { from -> out << from; }
+    }
+
+    def config = new File("$dest/config.zip").withOutputStream { out ->
+        new URL(configUrl).withInputStream { from -> out << from; }
+    }
+
+    def atlasFile = new ZipFile("$dest/atlas.zip")
+    atlasFile.entries().each { it ->
+        def path = Paths.get("$dest/atlas/" + it.name)
+        if(it.directory){
+            Files.createDirectories(path)
+        }
+        else {
+            def parentDir = path.getParent()
+            if (!Files.exists(parentDir)) {
+                Files.createDirectories(parentDir)
+            }
+            Files.copy(atlasFile.getInputStream(it), path)
+        }
+    }
+
+    def configFile = new ZipFile("$dest/config.zip")
+    configFile.entries().each { it ->
+        def path = Paths.get("$dest/config/" + it.name)
+        if(it.directory){
+            Files.createDirectories(path)
+        }
+        else {
+            def parentDir = path.getParent()
+            if (!Files.exists(parentDir)) {
+                Files.createDirectories(parentDir)
+            }
+            Files.copy(configFile.getInputStream(it), path)
+        }
+    }
+}
+
+workflow RBX {
+
+    take:
+        ch_fa               // channel: [ val(meta), [ fa ] ]
+        ch_tractogram       // channel: [ val(meta), [ tractogram ] ]
+        atlas_directory     // channel: [ directory ], optional: True
+
+    main:
+
+        ch_versions = Channel.empty()
+
+        // ** Setting up Atlas reference channels. ** //
+        if ( atlas_directory ) {
+            atlas_anat = Channel.fromPath("$atlas_directory/mni_masked.nii.gz")
+            atlas_config = Channel.fromPath("$atlas_directory/config_fss_1.json")
+            atlas_directory = Channel.fromPath("$atlas_directory/atlas/")
```

Comment:
I don't like this to be hardcoded.
Can we have something where you glob *json et *nii.gz
*json -> atlas_config
*nii.gz -> atlas_anat
atlas_directory is okay